packages/service-clients/end-to-end-tests/odsp-client/src/test/containerCreate.spec.ts

```diff
@@ -0,0 +1,160 @@
+		assert.strictEqual(typeof itemId, "string", "Attach did not return a string ID");
+	});
+
+	/**
+	 * Scenario: Test attaching a container.
+	 *
+	 * Expected behavior: an error should not be thrown nor should a rejected promise
+	 * be returned.
+	 */
+	it("can attach a container", async () => {
+		const { container } = await client.createContainer(schema);
+		const itemId = await container.attach();
+
+		if (container.connectionState !== ConnectionState.Connected) {
+			await timeoutPromise((resolve) => container.once("connected", () => resolve()), {
+				durationMs: connectTimeoutMs,
+				errorMsg: "container connect() timeout",
+			});
+		}
+
+		assert.strictEqual(typeof itemId, "string", "Attach did not return a string ID");
+		assert.strictEqual(
+			container.attachState,
+			AttachState.Attached,
+			"Container is not attached after attach is called",
+		);
+	});
+
+	/**
+	 * Scenario: Test if attaching a container twice fails.
+	 *
+	 * Expected behavior: an error should not be thrown nor should a rejected promise
+	 * be returned.
+	 */
+	it("cannot attach a container twice", async () => {
+		const { container } = await client.createContainer(schema);
+		const itemId = await container.attach();
+
+		if (container.connectionState !== ConnectionState.Connected) {
+			await timeoutPromise((resolve) => container.once("connected", () => resolve()), {
+				durationMs: connectTimeoutMs,
+				errorMsg: "container connect() timeout",
+			});
+		}
+
+		assert.strictEqual(typeof itemId, "string", "Attach did not return a string ID");
+		assert.strictEqual(
+			container.attachState,
+			AttachState.Attached,
+			"Container is attached after attach is called",
+		);
+		await assert.rejects(container.attach(), () => true, "Container should not attach twice");
+	});
+
+	/**
+	 * Scenario: test if Odsp Client can get an existing container.
+	 *
+	 * Expected behavior: an error should not be thrown nor should a rejected promise
+	 * be returned.
+	 */
+	it("can retrieve existing ODSP container successfully", async () => {
+		const { container: newContainer } = await client.createContainer(schema);
+		const itemId = await newContainer.attach();
+
+		if (newContainer.connectionState !== ConnectionState.Connected) {
+			await timeoutPromise((resolve) => newContainer.once("connected", () => resolve()), {
+				durationMs: connectTimeoutMs,
+				errorMsg: "container connect() timeout",
+			});
+		}
+
+		const resources = client.getContainer(itemId, schema);
+		await assert.doesNotReject(
+			resources,
+			() => true,
+			"container cannot be retrieved from ODSP",
+		);
+	});
+
+	/**
+	 * Scenario: test if Odsp Client can get a non-exiting container.
+	 *
+	 * Expected behavior: an error should be thrown when trying to get a non-existent container.
+	 */
+	it("cannot load improperly created container (cannot load a non-existent container)", async () => {
+		const consoleErrorFn = console.error;
+		console.error = (): void => {};
```

Comment:
why does this touch the console. we should never use console

Reply:
thanks for catching this. removed console